R/snapshot-reporter.R

```diff
@@ -101,6 +103,15 @@ SnapshotReporter <- R6::R6Class("SnapshotReporter",
       } else {
         snap_dir <- file.path(self$snap_dir, variant, self$file)
       }
+
+      save_path <- file.path(snap_dir, name)
+      if (save_path %in% self$snap_file_saved) {
+        cli::cli_abort(
+          "Snapshot file {.arg name} has already been saved. Please provide a unique snapshot file name."
+        )
+      }
+      self$snap_file_saved <- c(self$snap_file_saved, save_path)
```

Comment:
This is the only line I have a question on collecting the snap files already saved. 

It could be rewritten into 
```suggestion
      self$snap_file_saved[length(self$snap_file_saved) + 1] <- save_path
```
... but it is visually harder to read the line

(Maintainer's choice)